modules/raft-api/src/main/java/org/apache/ignite/internal/raft/rebalance/RaftCommandWithRetry.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.raft.rebalance;
+
+import static java.util.concurrent.CompletableFuture.failedFuture;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.apache.ignite.internal.raft.rebalance.ExceptionUtils.recoverable;
+import static org.apache.ignite.internal.util.CompletableFutures.copyStateTo;
+import static org.apache.ignite.lang.ErrorGroups.Common.NODE_STOPPING_ERR;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Function;
+import org.apache.ignite.internal.lang.IgniteInternalException;
+import org.apache.ignite.internal.lang.NodeStoppingException;
+import org.apache.ignite.internal.logger.IgniteLogger;
+import org.apache.ignite.internal.logger.Loggers;
+import org.apache.ignite.internal.util.CompletableFutures;
+import org.apache.ignite.internal.util.IgniteBusyLock;
+
+/**
+ * Helper class that executes raft command with retries.
+ */
+public class RaftCommandWithRetry {
+    private static final IgniteLogger LOG = Loggers.forClass(RaftCommandWithRetry.class);
+
+    private static final long MOVE_RESCHEDULE_DELAY_MILLIS = 100;
```

Comment:
We should add TODO here.
I'm not sure what exactly has to do with this number. Either avoid it after adding a possible infinity while waiting on the RAFT client, or add it to the configuration.